### PR TITLE
[MRG+1] MAINT moved "c" modules to pytest

### PR DESCRIPTION
**Pull request by NelleV**
Here is a first PR for the migration to pytest on all modules that start with a "c".

### CODE REVIEW

File: lib/matplotlib/tests/test_collections.py [hunk 1]
```diff
@@ -623,7 +621,7 @@ def test_pandas_indexing():
     try:
         import pandas as pd
     except ImportError:
-        raise SkipTest("Pandas not installed")
+        skip("Pandas not installed")
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_collections.py]: This one can be done as `pd = pytest.importorskip('pandas')`.

Comment (QuLogic) [inline lib/matplotlib/tests/test_collections.py]: Actually, now that I've looked into it further, `matplotlib.testing.skip` was added only in `master`, so I think we should definitely not add any more callers if we can. It should just go away once we've fully dropped nose.

File: lib/matplotlib/tests/test_colors.py [hunk 2]
```diff
@@ -609,17 +602,15 @@ def test_pandas_iterable():
     try:
         import pandas as pd
     except ImportError:
-        raise SkipTest("Pandas not installed")
-    if assert_sequence_equal is None:
-        raise SkipTest("nose lacks required function")
+        raise skip("Pandas not installed")
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_colors.py]: Same as above.

Comment (NelleV) [inline lib/matplotlib/tests/test_colors.py]: :(

**Comment by NelleV** [comment]
I much prefer assert_XX than the way pytest does it. It's much more readable.
I've removed all calls to matplotlib.testing's skip.

**Comment by anntzer** [comment]
@NelleV I'm curious why you think
```
assert_true(expr)
assert_equal(expr1, expr2)
```
is more readable than
```
assert expr
assert expr1 == expr2
```
Most editors will even highlight `assert` differently...

**Comment by NelleV** [comment]
@anntzer you know what you are testing straight at the beginning. With the latter option, it's harder to see what exactly you are testing. Basically, "explicit is better than implicit". Also, the error messages from nose.tools or numpy.testing are much more readable than the one you get from an assert.
As for syntax highlighting, some will argue that too much syntax highlighting is really bad to focus on code, and it's much better to have as little as possible (not sure I agree :) ).

I find our test suite in gerenal difficult to read. Both sklearn's and skimage's seem to me much much easier to work with. They both use less fancy features, less decorators, etc, yet these test suite are extremely easy to maintain.

**Comment by efiring** [comment]
As usual this is probably not the place for the discussion, but regardless, I will take the opportunity to agree with certain points made by @nellev:
1) I find some of the testing code very hard to understand--complex, obscure, minimally commented.
2) I don't see any major advantage one way or the other between `assert expr` and `assert_true(expr)`, but `assert_equal(a, b)` is vastly superior to `assert a == b`, because only the former can tell you what the mismatched values are.  This is critical when trying to figure out why a test failed.

**Comment by QuLogic** [comment]
Not when running in `pytest`; it prints out both sides of a comparison in an `assert`.

**Comment by efiring** [comment]
Thank you, that's good to know.  There's more magic going on than I realized.

**Comment by anntzer** [comment]
See e.g. http://pybites.blogspot.com/2011/07/behind-scenes-of-pytests-new-assertion.html.  Assertion rewriting is, IMO, *the* best feature of pytest.

**Comment by phobson** [comment]
Thanks @NelleV !

Backported to `v2.x` via 0a107f06a27c7d17f79e90966236de4f8add7b83

**Comment by NelleV** [comment]
Thanks @phobson for merging.
I see you've backported to 2.x, but I don't think we are backporting the pytest series to 2.x. @tacaswell can you confirm?

**Comment by QuLogic** [comment]
`master` _is_ 2.1; the `v2.x` branch was deleted. Please see @tacaswell's [message on the mailing list](https://mail.python.org/pipermail/matplotlib-devel/2017-January/000635.html). I have deleted the `v2.x` branch again.

**Pull request merged**